notebooks/03_evaluate/baseline_estimation.ipynb

```diff
@@ -0,0 +1,812 @@
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>MovieId</th>\n",
+       "      <th>Count</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>50</td>\n",
+       "      <td>426</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>100</td>\n",
+       "      <td>396</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>258</td>\n",
+       "      <td>390</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>181</td>\n",
+       "      <td>384</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>286</td>\n",
+       "      <td>363</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "   MovieId  Count\n",
+       "0       50    426\n",
+       "1      100    396\n",
+       "2      258    390\n",
+       "3      181    384\n",
+       "4      286    363"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "item_counts = train['MovieId'].value_counts().to_frame().reset_index()\n",
+    "item_counts.columns = ['MovieId', 'Count']\n",
+    "item_counts.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of user-item pairs: 1547463\n",
+      "After remove seen items: 1472463\n"
+     ]
+    }
+   ],
+   "source": [
+    "user_item_col = ['UserId', 'MovieId']\n",
+    "\n",
+    "# Cross join users and items\n",
+    "test_users = test['UserId'].unique()\n",
+    "user_item_list = list(itertools.product(test_users, item_counts['MovieId']))\n",
+    "users_items = pd.DataFrame(user_item_list, columns=user_item_col)\n",
+    "\n",
+    "print(\"Number of user-item pairs:\", len(users_items))\n",
+    "\n",
+    "# Remove seen items (items in the train set) as we will not recommend those again to the users\n",
+    "users_items_remove_seen = users_items.loc[\n",
+    "    ~users_items.set_index(user_item_col).index.isin(train.set_index(user_item_col).index)\n",
```

Comment:
How about `~users_items[user_item_col].isin(train[user_item_col].tolist())` instead. It is a bit shorter.